adapter-excel/src/main/java/ch/prevo/open/node/adapter/excel/ExcelReader.java

```diff
@@ -75,32 +73,87 @@
             if (wb != null) {
-                employmentTerminations = mapRows(wb.getSheetAt(0), this::mapEmploymentTermination);
+                employmentTerminations = mapRows(wb.getSheet(TERMINATION_LABEL), this::mapEmploymentTermination);
             }
         } catch (IOException e) {
             LOGGER.error("An exception occurred while trying to read the employment terminations", e);
         }
         return employmentTerminations;
     }
 
+    @Override
+    public void notifyTerminationErrors(Map<EmploymentTermination, Set<ConstraintViolation<EmploymentTermination>>> violations) {
+        final String filename = getErrorFilename();
+        final Workbook workbook = getErrorWorkbook(filename);
+
+        if (workbook != null) {
+            try (final ErrorWriter writer = new ErrorWriter(workbook, filename)) {
+                writer.markTerminationErrors(violations);
+            } catch (IOException e) {
+                LOGGER.error("An exception occurred while trying to read the employment terminations", e);
+            }
+        }
+    }
+
     @Override
     public List<EmploymentCommencement> getEmploymentCommencements() {
         List<EmploymentCommencement> employmentCommencements = Collections.emptyList();
         try (final Workbook wb = getWorkbook()) {
             if (wb != null) {
-                employmentCommencements = mapRows(wb.getSheetAt(1), this::mapEmploymentCommencement);
+                employmentCommencements = mapRows(wb.getSheet(COMMENCEMENTS_LABEL), this::mapEmploymentCommencement);
             }
         } catch (IOException e) {
             LOGGER.error("An exception occurred while trying to read the employment commencements", e);
         }
         return employmentCommencements;
     }
 
+    @Override
+    public void notifyCommencementErrors(Map<EmploymentCommencement, Set<ConstraintViolation<EmploymentCommencement>>> violations) {
+        final String filename = getErrorFilename();
+        final Workbook workbook = getErrorWorkbook(filename);
+
+        if (workbook != null) {
+            try (final ErrorWriter writer = new ErrorWriter(workbook, filename)) {
+                writer.markCommencementErrors(violations);
+            } catch (IOException e) {
+                LOGGER.error("An exception occurred while trying to read the employment terminations", e);
+            }
+        }
+    }
+
+    private Workbook getErrorWorkbook(String filename) {
+        final Path path = Paths.get(filename);
+        if (Files.exists(path)) {
+            try (final InputStream inputStream = new FileInputStream(filename)) {
+                return WorkbookFactory.create(inputStream);
+            } catch (IOException | EncryptedDocumentException | EmptyFileException | InvalidFormatException e) {
+                LOGGER.warn("Unable to read existing error file {}, creating a new one", filename);
+                try {
+                    Files.delete(path);
```

Comment:
I think in FileUtils of apache commons is a `FileUtil.deleteQuietly` https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#deleteQuietly(java.io.File)

Reply:
Right, thanks.